src/docs/codelabs/extended-flex-widget.md

```diff
@@ -0,0 +1,597 @@
+{{site.alert.tip}}
+  Before moving to the next code example, change the `flex` properties to other values, 
+  such as 2 and 1.
+{{site.alert.end}}
+
+#### Example: Testing fit properties
+{:.no_toc}  
+{{site.alert.secondary}}
+  The following example displays the `fit` property,
+  which can have one of two values:
+
+  `FlexFit.loose`
+  : The widget's preferred size is used. (Default)
+
+  `FlexFit.tight`
+  : Forces the widget to fill all of its extra space. 
+
+  In this example, modify the `fit` properties to
+  make the `Flexible` widgets fill the free space.
+
+  **1.** Click the **Run** button. 
+
+  **2.** Change the `fit` values to `FlexFit.tight`, 
+  and run again.   
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/ba0f40356d1023066d960f6de2be1a4b
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=ba0f40356d1023066d960f6de2be1a4b&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## Expanded widget
+
+Similar to the `Flexible` widget, the `Expanded` widget can wrap a widget and force the widget to fill extra space. 
+
+{{site.alert.tip}}
+  **What's the difference between Flexible and Expanded?**
+    Use `Flexible` to resize widgets in a `Row` or `Column`,
+    so you can adjust child widgets' spacing while keeping the
+    relation with their parent widget.
+    `Expanded` changes the constraints of the child widgets
+    to fill any empty space.
+{{site.alert.end}} 
+
+#### Example: Filling extra space
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example demonstrates how the `Expanded` widget forces its child widget to fill extra space.
+
+  **1.** Click the **Run** button. 
+
+  **2.** Wrap the second `BlueBox` widget in an `Expanded` widget. 
+
+  For example: 
+
+  ```dart
+  Expanded(child: BlueBox(),),
+  ```
+  **3.** Select the **Format** button, and run again. 
+
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/77021d2ed15f9ece850de15e73c47526
+{% endcomment %}
+<iframe src="https://dartpad.dev/experimental/embed-new-flutter.html?id=77021d2ed15f9ece850de15e73c47526&amp;theme=dark&amp;split=60" width="100%" height="400px"></iframe>
+
+## SizedBox widget
+
+The `SizedBox` widget can be used in one of two ways.
+When `SizedBox` wraps a widget, 
+it resizes the widget using the `height` and `width` properties.
+When it doesn't wrap a widget, 
+it uses the `height` and `width` properties to create empty space.
+#### Example: Resizing a widget 
+{:.no_toc}
+{{site.alert.secondary}}
+  The following example wraps the middle `BlueBox` widget inside of a
+  `SizedBox` widget and sets the `BlueBox`'s width to 100 logical pixels.
+
+  **1.** Click the **Run** button.
+
+  **2.** Add a `height` property equal to 100 logical pixels inside the `SizedBox` widget, and run again.   
+{{site.alert.end}}
+{% comment %}
+  Gist: https://gist.github.com/6582851e85b57180ff5321f814fabb81
```

Comment:
The solution uses the wrong main axis alignment. Instead of spaceAround, it should be start, end, or center.

```
mainAxisAlignment: MainAxisAlignment.spaceAround,
```

The test is wrong. The text says to use start, end, or center, but it fails if you use start, and it doesn't fail if you use spaceAround.

```
  if (row.mainAxisAlignment == MainAxisAlignment.start
      || row.mainAxisAlignment == MainAxisAlignment.spaceBetween 
      || row.mainAxisAlignment == MainAxisAlignment.spaceEvenly) {
    _result(false, ['It\'s best to use MainAxisAlignment.start, MainAxisAlignment.end, or MainAxisAlignment.center to see how the SizedBox widgets work in a Row.']);
    return;
  }
```